roles/openshift_health_checker/openshift_checks/docker_image_availability.py

```diff
@@ -55,59 +65,85 @@ def required_images(self, task_vars):
         deployment_type = get_var(task_vars, "deployment_type")
         image_base_name = self.docker_image_base[deployment_type]
 
-        # openshift_release = get_var(task_vars, "openshift_release")
-        # openshift_image_tag = get_var(task_vars, "openshift_image_tag")
-        openshift_release = "v3.4"
-        openshift_image_tag = "3.4.1.7"
+        openshift_release = get_var(task_vars, "openshift_release")
+        openshift_image_tag = get_var(task_vars, "openshift_image_tag")
 
         is_containerized = get_var(task_vars, "openshift", "common", "is_containerized")
 
         if is_containerized:
-            docker_or_rpm_images = set(self.containerized_docker_images(image_base_name, openshift_release))
+            images = set(self.containerized_docker_images(image_base_name, openshift_release))
         else:
-            docker_or_rpm_images = set(self.rpm_docker_images(image_base_name, openshift_release))
+            images = set(self.rpm_docker_images(image_base_name, openshift_release))
 
         # append images with qualified image tags to our list of required images.
         # these are images with a (v0.0.0.0) tag, rather than a standard release
         # format tag (v0.0). We want to check this set in both containerized and
         # non-containerized installations.
-        docker_or_rpm_images.update(self.qualified_docker_images(self.image_from_base_name(image_base_name),
-                                                                 "v" + openshift_image_tag))
+        images.update(
+            self.qualified_docker_images(self.image_from_base_name(image_base_name), "v" + openshift_image_tag)
+        )
 
-        return docker_or_rpm_images
+        return images
 
     def local_images(self, required_images, task_vars):
-        """ Return all found local docker images """
-        found = set()
-        for image in required_images:
-            cmd = self.module_executor("docker_image_facts", {"name": image}, task_vars)
-            if cmd.get("images", None):
-                found.add(image)
+        """Filter a list of images and return those available locally."""
+        return [
+            image for image in required_images
+            if self.is_image_local(image, task_vars)
+        ]
 
-        return found
+    def is_image_local(self, image, task_vars):
+        result = self.module_executor("docker_image_facts", {"name": image}, task_vars)
+        if result.get("failed", False):
+            return False
+
+        return bool(result.get("images", []))
 
     def known_docker_registries(self, task_vars):
-        docker_info = self.module_executor("docker_info", {}, task_vars).get("info", "")
+        result = self.module_executor("docker_info", {}, task_vars)
+
+        # do test on module exception - see if this line is useful
```

Comment:
> Did you have time to check it?

Yes, at least when a call to the docker `client.info()` results in a `404` response from the daemon, the following is returned by the `docker_info` module:

```
http://pastebin.test.redhat.com/467479
```

Both `rc=1` and `failed=True` are present key-values in the response:

```
...
'failed': True,
                                                       'module_stderr': u'Traceback (most recent call last):\n  File "/tmp/ansible_mKSMn8/ansible_module_docker_info.py", line 24, in <module>\n    main()\n  File "/tmp/ansible_mKSMn8/ansible_module_docker_info.py", line 19, in main\n    info=client.info(),\n  File "/usr/lib/python2.7/site-packages/docker/api/daemon.py", line 33, in info\n    return self._result(self._get(self._url("/inf")), True)\n  File "/usr/lib/python2.7/site-packages/docker/client.py", line 178, in _result\n    self._raise_for_status(response)\n  File "/usr/lib/python2.7/site-packages/docker/client.py", line 173, in _raise_for_status\n    raise errors.NotFound(e, response, explanation=explanation)\ndocker.errors.NotFound: 404 Client Error: Not Found ("{"message":"page not found"}")\n',
                                                       'module_stdout': u'',
                                                       'msg': u'MODULE FAILURE',
                                                       'rc': 1}}}
```